### PR TITLE
Add support import.meta.env

### PR DESCRIPTION
**Pull request by otgerrogla**
### What does this PR do?

Implements #4667. Adds support for `import.meta.env` as an alias to `process.env` or `Bun.env`, to improve compatibility with `vite` and other tools.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

I wrote automated tests

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
I wonder if we need to do this at build time too.

File: src/bun.js/bindings/ImportMetaObject.cpp [hunk 1]
```diff
@@ -412,6 +417,7 @@ static const HashTableValue ImportMetaObjectPrototypeValues[] = {
     { "file"_s, static_cast<unsigned>(JSC::PropertyAttribute::ReadOnly | JSC::PropertyAttribute::CustomAccessor | PropertyAttribute::DontDelete), NoIntrinsic, { HashTableValue::GetterSetterType, jsImportMetaObjectGetter_file, 0 } },
     { "path"_s, static_cast<unsigned>(JSC::PropertyAttribute::ReadOnly | JSC::PropertyAttribute::CustomAccessor | PropertyAttribute::DontDelete), NoIntrinsic, { HashTableValue::GetterSetterType, jsImportMetaObjectGetter_path, 0 } },
     { "require"_s, static_cast<unsigned>(JSC::PropertyAttribute::ReadOnly | JSC::PropertyAttribute::CustomAccessor | PropertyAttribute::DontDelete), NoIntrinsic, { HashTableValue::GetterSetterType, jsImportMetaObjectGetter_require, 0 } },
+    { "env"_s, static_cast<unsigned>(JSC::PropertyAttribute::ReadOnly | JSC::PropertyAttribute::CustomAccessor | PropertyAttribute::DontDelete), NoIntrinsic, { HashTableValue::GetterSetterType, jsImportMetaObjectGetter_env, 0 } },
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/ImportMetaObject.cpp:420]: Is it read-only in Vite?

File: src/bun.js/bindings/ImportMetaObject.cpp [hunk 2]
```diff
@@ -403,6 +403,11 @@ JSC_DEFINE_CUSTOM_GETTER(jsImportMetaObjectGetter_require, (JSGlobalObject * glo
 
     return JSValue::encode(thisObject->requireProperty.getInitializedOnMainThread(thisObject));
 }
+JSC_DEFINE_CUSTOM_GETTER(jsImportMetaObjectGetter_env, (JSGlobalObject * jsGlobalObject, JSC::EncodedJSValue thisValue, PropertyName propertyName))
+{
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/ImportMetaObject.cpp:407]: Can you add a test that runs this in node:vm? Just to make sure this jsCast doesn’t crash when the global object is not a Zig::GlobalObject

**Comment by Jarred-Sumner** [comment]
Thank you for this. Couple questions, but it’s merge-able so I’m merging it.

File: src/bun.js/bindings/ImportMetaObject.cpp (hunk 2 above)
Comment (otgerrogla) [reply src/bun.js/bindings/ImportMetaObject.cpp:407]: Tried to run this test, but the parser complains with `SyntaxError: import.meta is only valid inside modules.`, and I couldn't find any way to force it to run as an ESM, other than using node:vm modules which are not yet implemented.

File: src/bun.js/bindings/ImportMetaObject.cpp (hunk 1 above)
Comment (otgerrogla) [reply src/bun.js/bindings/ImportMetaObject.cpp:420]: The property of the env map/object itself (`Object.getOwnPropertyDescriptor(import.meta, 'env')`) I'd say so.
The values inside of the map can be writen to, I just checked it

File: src/bun.js/bindings/ImportMetaObject.cpp (hunk 2 above)
Comment (Jarred-Sumner) [reply src/bun.js/bindings/ImportMetaObject.cpp:407]: you could pass the import.meta.env object to the node:vm instance

**Pull request merged**